### ISSUE TITLE
.append() wildcard support?

### ISSUE THREAD

**Issue by bondarewicz** [OctopusDeploy/octopackjs#13]
Does this package support wildcards?

I have a following code

```
var octo = require('@octopusdeploy/octopackjs');

console.log('Creating octopack version:', process.argv[process.argv.length-1]);
var v = process.argv[process.argv.length-1];
var options = {
  version: v
}

octo.pack('zip', options).append('./build/*.js').toFile('./octo', function (err, data) {
  console.log('Package ' + data.name + ' created in ' + data.path);
});
```

I wonder if there is any way I could use`.append('./build/*.js')` or better `.append('./build/**')` or similar ? I basically need to grab all the files from the build folder and use them for octopack

**Comment by DmitryFillo** [comment]
Seems like it doesn't support wildcards. But there is another plugin: [gulp-octo](https://github.com/OctopusDeploy/gulp-octo). It's just a wrapper for octopackjs with gulp support. I've tried it too, works good with wildcards, but fails with subdirectories, also gulp.dest doesn't work correct with this. I haven't debugged it well enough, becuase I've found simplier solution for packing:

```
const fs = require('fs');
const gulp = require('gulp');
const zip = require('gulp-zip');

/**
 * Get package.json data.
 *
 * @returns {Object} package.json parsed data.
 */
const getPackageJSONData = () => {
    return JSON.parse(fs.readFileSync('./package.json', 'utf8'));
};

/**
 * Get Octopus package data.
 *
 * @param {Object} packageData package.json parsed data.
 * @returns {{name: string, version: string}} Octopus package data.
 */
const getOctopusPackageData = (packageData) => {
    return {
        name: packageData.name.replace('/', '.').replace('@',''),
        version: packageData.version,
    };
};

/**
 * Get package name for packing with octopus.
 *
 * @param {Object} octoData Octopus package data.
 * @returns {string} Octopus package name.
 */
const getOctopusPackageName = (octoData) => {
    return `${octoData.name}.${octoData.version}.zip`;
};

gulp.task('pack', () => {
    const packageData = getPackageJSONData();
    const octoData = getOctopusPackageData(packageData);
    const packageName = getOctopusPackageName(octoData);

    gulp.src(['dist/**/*',])
        .pipe(zip(packageName))
        .pipe(gulp.dest('./bin'));
});
```

For pushing you can use octopackjs, of course. It works for my cases.

**Comment by bondarewicz** [comment]
@DmitryFillo thanks, I'm aware of gulp-octo and I did wanted to avoid it specifically as build process depends only on npm, anyways .. due to lack of wildcard support I simply replaced it with following:

```
var zipdir = require('zip-dir');
var fs = require('fs');
var dir = './octo';
var ver = process.argv[process.argv.length-1];

if (!fs.existsSync(dir)) fs.mkdirSync(dir);

zipdir('build', {
    each: path => console.log(path, "added!"),
    saveTo: dir + '/project-octopack.'+ ver +'.zip'
  }, function (err, buffer) {
  if(err){
    console.error(err);
    process.exit(1);
  } else {
    console.log('all done!');
  }
});
```

I did not needed a package push as this was handled by CI itself.

**Comment by zentron** [comment]
globs are now available in `0.1.0`
